src/service/viewer-impl.js

```diff
@@ -742,7 +742,7 @@ export class Viewer {
    * Triggers "documentLoaded" event for the viewer.
    */
   postDocumentReady() {
-    this.sendMessageUnreliable_('documentLoaded', {
+    this.sendMessageUnreliable('documentLoaded', {
```

Comment:
I'd like to reconsider these calls in an alternative way. We already have a very clear idea whether or not (a) doc is embedded; and (b) where the messaging is expected. With this in mind, I'd rather change this code to either wait for messaging ready and reliably send this message or not even try if messaging is not coming. We currently have, I believe, `maybeSendMessage_` that does something like that in Viewer.

Some code in "sendMessageUnreliable" has a useful feature to de-dupe the messages. But it's too coarse and doesn't work very well for all cases. I'd rather move deduplication, if needed, to the actual class that issues messages.

WDYT? So, the bottom line, the `sendMessageUnreliable_` will be completely gone.